src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -195,6 +212,417 @@ TEST_F(
+
+  // check: return value is false and list is empty
+  EXPECT_FALSE(result);
+  EXPECT_TRUE(rejected_params.empty());
+}
+
+TEST_F(ApplicationManagerImplTest,
+       OnServiceStartedCallback_VideoServiceNotStart2) {
+  AddMockApplication();
+
+  const connection_handler::DeviceHandle device_handle = 0;
+  const protocol_handler::ServiceType service_type =
+      protocol_handler::ServiceType::kMobileNav;
+  const int32_t session_key = 123;
+  EXPECT_CALL(*mock_app_ptr_, app_id()).WillRepeatedly(Return(session_key));
+  EXPECT_CALL(*mock_app_ptr_, is_navi()).WillRepeatedly(Return(true));
+  // HMI level is not FULL nor LIMITED
+  EXPECT_CALL(*mock_app_ptr_, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+
+  bool result = false;
+  std::vector<std::string> rejected_params;
+  EXPECT_CALL(mock_connection_handler_, NotifyServiceStartedResult(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), SaveArg<1>(&rejected_params)));
+
+  // check: SetVideoConfig() and StartStreaming() should not be called
+  EXPECT_CALL(*mock_app_ptr_, SetVideoConfig(_, _)).Times(0);
+  EXPECT_CALL(*mock_app_ptr_, StartStreaming(service_type)).Times(0);
+
+  app_manager_impl_->OnServiceStartedCallback(
+      device_handle, session_key, service_type, NULL);
+
+  // check: return value is false and list is empty
+  EXPECT_FALSE(result);
+  EXPECT_TRUE(rejected_params.empty());
+}
+
+TEST_F(ApplicationManagerImplTest,
+       OnServiceStartedCallback_VideoSetConfig_SUCCESS) {
+  AddMockApplication();
+
+  const connection_handler::DeviceHandle device_handle = 0;
+  const protocol_handler::ServiceType service_type =
+      protocol_handler::ServiceType::kMobileNav;
+  const int32_t session_key = 123;
+  EXPECT_CALL(*mock_app_ptr_, app_id()).WillRepeatedly(Return(session_key));
+  EXPECT_CALL(*mock_app_ptr_, is_navi()).WillRepeatedly(Return(true));
+  EXPECT_CALL(*mock_app_ptr_, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_LIMITED));
+
+  bool result = false;
+  std::vector<std::string> rejected_params;
+  EXPECT_CALL(mock_connection_handler_, NotifyServiceStartedResult(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), SaveArg<1>(&rejected_params)));
+
+  BsonObject input_params;
+  bson_object_initialize_default(&input_params);
+  char protocol_version[] = "1.0.0";
+  bson_object_put_string(&input_params, "protocolVersion", protocol_version);
+  bson_object_put_int64(&input_params, "mtu", 100);
+  char protocol_name[] = "RTP";
+  bson_object_put_string(&input_params, "videoProtocol", protocol_name);
+  char codec_name[] = "VP9";
+  bson_object_put_string(&input_params, "videoCodec", codec_name);
+  bson_object_put_int32(&input_params, "desiredHeight", 640);
+  bson_object_put_int32(&input_params, "desiredWidth", 480);
+
+  smart_objects::SmartObject converted_params(smart_objects::SmartType_Map);
+  converted_params["protocol"] = hmi_apis::Common_VideoStreamingProtocol::RTP;
```

Comment:
Done in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/229a2901e652be32d39e5d8559fb05f1a794dde5.